pkg/apis/apps/validation/validation.go

```diff
@@ -108,6 +108,25 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path, op
 					apps.OnDeleteStatefulSetStrategyType)))
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) {
+		requiredError := field.Required(fldPath.Child("persistentVolumeClaimDeletePolicy"), "")
+		if spec.PersistentVolumeClaimDeletePolicy == nil {
+			allErrs = append(allErrs, requiredError)
+		} else {
+			switch *spec.PersistentVolumeClaimDeletePolicy {
+			case "":
+				allErrs = append(allErrs, requiredError)
+			case apps.RetainPersistentVolumeClaimDeletePolicyType, apps.DeleteOnStatefulSetDeletionOnlyPersistentVolumeClaimDeletePolicyType, apps.DeleteOnScaledownOnlyPersistentVolumeClaimDeletePolicyType, apps.DeleteOnScaledownAndStatefulSetDeletionPersistentVolumeClaimDeletePolicyType:
+			default:
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("persistentVolumeClaimDeletePolicy"), spec.PersistentVolumeClaimDeletePolicy, fmt.Sprintf("must be '%s', '%s', '%s' or '%s'", apps.RetainPersistentVolumeClaimDeletePolicyType, apps.DeleteOnStatefulSetDeletionOnlyPersistentVolumeClaimDeletePolicyType, apps.DeleteOnScaledownOnlyPersistentVolumeClaimDeletePolicyType, apps.DeleteOnScaledownAndStatefulSetDeletionPersistentVolumeClaimDeletePolicyType)))
+			}
+		}
+	} else {
+		if spec.PersistentVolumeClaimDeletePolicy != nil {
```

Comment:
For rollback scenarios, we should not validate that the field is unset (trust that it gets dropped appropriately in the strategy). If the field was already set before rollback, we continue letting the field be, so that the object will not be blocked on deletion/update.

Reply:
Yeah, I figured that out. I now validate if the field != nil, regardless of the feature gate.